kore/src/Kore/IndexedModule/Resolvers.hs

```diff
@@ -340,3 +320,88 @@ findIndexedSort
+
+
+{- Utilities -}
+
+-- It would make sense to put this in a `where` clause; however,
+-- the fully type annotation is required even there, and that makes
+-- for too much clutter.
+applyToSentence
+    :: (MetaOrObject level)
+    => (forall ssoa .  SentenceSymbolOrAlias ssoa
+       => [Sort level]
+       -> ssoa level pat
+       -> result)
+    -> IndexedModule param pat declAtts axiomAtts
+    -> SymbolOrAlias level
+    -> result
+applyToSentence f =
+     applyToResolution (\ params (_, sentence) -> f params sentence)
+
+-- It would make sense to put this in a `where` clause; however,
+-- the fully type annotation is required even there, and that makes
+-- for too much clutter.
+applyToAttributes
+    :: MetaOrObject level
+    => (declAtts -> result)
+    -> IndexedModule sortParam patternType declAtts axiomAtts
+    -- ^ module representing an indexed definition
+    -> SymbolOrAlias level     -- ^the head we want to find sorts for
+    -> result
+applyToAttributes f =
+    applyToResolution (\ _ (attrs, _) -> f attrs)
+
+
+applyToResolution :: (HasCallStack, MetaOrObject level)
+                   => (forall ssoa .  SentenceSymbolOrAlias ssoa
+                       => [Sort level]
+                       -> (declAtts, ssoa level pat)
+                       -> result)
+                   -> IndexedModule param pat declAtts axiomAtts
+                   -> SymbolOrAlias level
+                   -> result
+applyToResolution f m patternHead =
+    case symbolResult <> aliasResult of
+        Right result -> result
+        Left _ -> error $ noHead patternHead
+  where
+    headName = symbolOrAliasConstructor patternHead
+    headParams = symbolOrAliasParams patternHead
+    symbolResult = f headParams <$> resolveSymbol m headName
+    aliasResult = f headParams <$> resolveAlias m headName
+
+{- Support for warning of undefined values -}
+
+-- | `error` with a helpful message in case of `Left`.
+-- | Otherwise, return what `Right` returns.
+assertRight :: Either (Error err) desired -> desired
+assertRight wrapped =
+    case wrapped of
+        Left err      -> error (printError err)
+        Right desired -> desired
+
+-- | A message declaring that a Sort is undefined
+noSort :: MetaOrObject level => Id level -> String
+noSort sortId =
+    notDefined "Sort" $ show sortId
+
+-- | A message declaring that a Head is undefined
+noHead :: MetaOrObject level => SymbolOrAlias level -> String
+noHead patternHead =
+    notDefined "Head" $ show patternHead
+
+-- | A message declaring that a Alias is undefined
+noAlias :: MetaOrObject level => Id level -> String
+noAlias identifier =
+    notDefined "Alias" $ getIdForError identifier
+
+-- | A message declaring that a Symbol is undefined
+noSymbol :: MetaOrObject level => Id level -> String
+noSymbol identifier =
+    notDefined "Symbol" $ getIdForError identifier
+
+-- | A message declaring that some `tag` is undefined.
+notDefined :: String -> String -> String
```

Comment:
This would be nice if we could catch and handle these errors, but I don't think it's necessary in this case: if there is an error, `assertRight` brings the program to a crashing halt anyway!

> make `assertRight` take a `PrettyPrint e => Either e desired -> desired`

This would be nice-to-have, but as a reformed gradualist, I think it's fine if we don't address it now. :grinning: